src/autopas/containers/cellPairTraversals/CSlicedBasedTraversal.h

```diff
@@ -0,0 +1,235 @@
+/**
+ * @file CSlicedBasedTraversal.h
+ *
+ * @date 24 May 2020
+ * @author fischerv
+ */
+
+#pragma once
+
+#include <algorithm>
+
+#include "autopas/containers/cellPairTraversals/CellPairTraversal.h"
+#include "autopas/utils/DataLayoutConverter.h"
+#include "autopas/utils/ThreeDimensionalMapping.h"
+#include "autopas/utils/WrapOpenMP.h"
+
+namespace autopas {
+
+/**
+ * This class provides the colored sliced traversal.
+ *
+ * The traversal finds the longest dimension of the simulation domain and cuts
+ * the domain in one slice (block) per thread along this dimension. Unlike the regular
+ * sliced traversal, this version uses a 2-coloring to prevent race conditions, instead of
+ * locking the starting layers. This could also be describes as a c02-traversal. This class
+ * is however not derived from CBasedTraversal, as that would not allow varying slice thicknesses,
+ * and would prevent us from selecting the dimension in which we cut the slices.
+ *
+ * @tparam ParticleCell The type of cells.
+ * @tparam PairwiseFunctor The functor that defines the interaction of two particles.
+ * @tparam dataLayout
+ * @tparam useNewton3
+ */
+template <class ParticleCell, class PairwiseFunctor, DataLayoutOption::Value dataLayout, bool useNewton3>
+class CSlicedBasedTraversal : public CellPairTraversal<ParticleCell> {
```

Comment:
this class seems to have some overlap to `SlicedBasedTraversal`. Maybe a common base class or one inheriting from the other would be reasonable to reduce code duplication.